### DISCUSSION TITLE
Question about including variables in model.rhs

### DISCUSSION THREAD

**Discussion by YouHyin** [pybamm-team/PyBaMM#2850] [Q&A]
I want to set the discharge capacity limit. (ex. Charge at 1C until 0.88 A.h)
So I added this code in def update_new_model_events.(https://github.com/pybamm-team/PyBaMM/blob/main/pybamm/simulation.py)
        if "Capacity cut-off [A.h]" in op:
            new_model.events.append(
                pybamm.Event(
                    "Capacity cut-off (CCCV) [A] [experiment]",
                    (
                            new_model.variables["Discharge capacity [A.h]"]
                            < (pybamm.InputParameter("Capacity input [A.h]") - 1e-4)
                    ),
                )
            )

![주석 2023-04-03 194418](https://user-images.githubusercontent.com/113442395/229487587-9e4eb3a5-c296-47cb-9a20-91622e20d044.png)

I want to include variable 'Discharge capacity [A.h]' in model.rhs or model.algebraic.
However, I don't know which code to add in model.rhs. 
Should I add in basemodel.py ??(https://github.com/pybamm-team/PyBaMM/blob/main/pybamm/models/base_model.py)

Thank you!

**Comment by RuiheLi** [comment]
Hi, I think this may be linked to the fact that the model in `model.rhs` is not the same model and by the time it is created, it doesn't contain the variables we need, i.e., "Discharge capacity [A.h]", the key script may be [this](https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/simulation.py#:~:text=%23%20Create%20a%20new%20model%20for,(new_model%2C%20op)). No sure how to solve this, would you have some quick ideas? @brosaplanella
